libs/feature/editor/src/lib/components/record-form/form-field/form-field-map-container/form-field-map-container.component.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core'
+import { CommonModule } from '@angular/common'
+import { catchError, from, map, Observable, of, switchMap } from 'rxjs'
+import {
+  DEFAULT_BASELAYER_CONTEXT,
+  FeatureMapModule,
+  MapContextLayerTypeEnum,
+  MapContextModel,
+  MapFacade,
+  MapStyleService,
+  MapUtilsService,
+} from '@geonetwork-ui/feature/map'
+import { Extent } from 'ol/extent'
+import { Fill, Stroke, Style } from 'ol/style'
+import { getOptionalMapConfig, MapConfig } from '@geonetwork-ui/util/app-config'
+import { FeatureCollection, Geometry } from 'geojson'
+import { GeoJSONFeatureCollection } from 'ol/format/GeoJSON'
+import {
+  DatasetSpatialExtent,
+  Keyword,
+} from '@geonetwork-ui/common/domain/model/record'
+import { Polygon } from 'ol/geom'
+import GeoJSON from 'ol/format/GeoJSON'
+
+@Component({
+  selector: 'gn-ui-form-field-map-container',
+  standalone: true,
+  imports: [CommonModule, FeatureMapModule],
+  templateUrl: './form-field-map-container.component.html',
+  styleUrls: ['./form-field-map-container.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FormFieldMapContainerComponent implements OnChanges {
+  @Input() keywordsWithSpatialExtents: {
+    [key: string]: {
+      placeKeyword?: Keyword
+      spatialExtents?: DatasetSpatialExtent
+    }
+  }[]
```

Comment:
Not critical, but this component doesn't need to know anything about keywords, so it would be better if you only gave it an array of spatial extents as input. This could be done by reducing/flattening your keywordsWithSpatialExtents from the parent component before sending it to the child component.